cmd/run/run_test.go

```diff
@@ -276,13 +276,12 @@ func TestBuildServiceWithPresharedKeyAuthentication(t *testing.T) {
 
 func TestBuildServiceWithTracingEnabled(t *testing.T) {
 	// create mock OTLP server
-	otlpServerPort, otlpServerPortReleaser := TCPRandomPort()
+	otlpServerPort := testutils.TCPRandomPort()
 	localOTLPServerURL := fmt.Sprintf("localhost:%d", otlpServerPort)
-	otlpServerPortReleaser()
```

Comment:
we don't need to have `TCPRandomPort` return a function that stops listening on the port, the function can stop listening automatically.